frontend/src/pages/question/questionBuilder/questionTemplate/QuestionTemplate.jsx

```diff
@@ -44,19 +48,24 @@ function displayQuestion(questionName) {
 
 export default function QuestionTemplateModal(props) {
   const {
-    selectQuestion, showModal, questions, concepts, close,
+    selectQuestion, showModal, close,
   } = props;
   const [questionTemplate, setQuestionTemplate] = useState({});
   const [questionName, updateQuestionName] = useState([]);
-  const [disableSubmit, setDisableSubmit] = useState(true);
   const [nameList, updateNameList] = useState([]);
   const [types, setTypes] = useState([]);
   const [labels, setLabels] = useState([]);
   const [curies, setCuries] = useState([]);
-  const curieSelector = useRef({});
 
-  function setFocus(num) {
-    curieSelector.current[`curie${num}`].curieSelector.input.focus();
+  // Using a ref here ensures that we have an up-to-date nameList value
+  // in the setFocus method
+  const nameListRef = useRef(null);
+  nameListRef.current = nameList;
```

Comment:
You can initialize a ref with a default value on the line above instead of giving it null.